File: 1051.height-checker.py
#
# @lc app=leetcode id=1051 lang=python3
#
# [1051] Height Checker
#

# @lc code=start
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        total = 0
        for i,n in enumerate(sorted(heights)):
            if heights[i] != n:
                total += 1

        return total
# @lc code=end

File: 2020/1051.height-checker.js
/*
 * @lc app=leetcode id=1051 lang=javascript
 *
 * [1051] Height Checker
 */

// @lc code=start
/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function(heights) {
  let sorted = heights.slice().sort((a,b)=>a-b)
  let count = []
  for (let i=0;i<heights.length;i++){
    if (heights[i] !== sorted[i]) {
      let swapIdx = heights.lastIndexOf(sorted[i])
      let swap = heights[swapIdx]
      heights[swapIdx] = heights[i]
      heights[i] = swap
      count[i] = true
      count[swapIdx] = true
    }
  }
  return count.filter(v=>v===true).length
};
// @lc code=end

